[CleanUp] Add fully qualified names for std::string and std::map

mrtineide opened the pull request:
## Namespace cleanup - Change in other libraries will affect O2 

## Story of what happened 

I was contacted by @costing, who relayed a report from @vkucera regarding problematic headers. It was discovered while running `clang-tidy` on `O2Physics`. The issue was that while running `cland-tidy` it would include the header `TJAlienCredentials.h`  from the `JAlien-ROOT` or `libjalienO2` libraries instead of `#include <string>`.

Specifically, the header contained the following lines:

```cpp
using std::map;
using std::string;
```

These `using` declarations in a header file pollute the global namespace for every file that uses them. A common place where this was spread was through `CCDBApi.h`, which includes `TJAlienCredentials.h`, thereby spreading the `using std::string` declaration across many parts that uses CCDB API.

I am new to C++ so I had to find some sources that confirmed why the change should be considered:
- [ALICE Coding Guidelines)](https://rawgit.com/AliceO2Group/CodingGuidelines/master/coding_guidelines.html?showone=Using_declarations_and_directives#Using_declarations_and_directives)
- [LLVM Coding Standards](https://llvm.org/docs/CodingStandards.html#do-not-use-using-namespace-std)
- [LearnCpp.com on Using Declarations](https://www.learncpp.com/cpp-tutorial/using-declarations-and-using-directives/)

### The proposed change in the other repo's:
**In external repos**: Removed `using std::string` and `using std::map` from `TJAlienCredentials.h`  
  - [Merge request to JAlien-ROOT](https://gitlab.cern.ch/jalien/jalien-root/-/merge_requests/88)
  - [Merge request to libjalienO2](https://gitlab.cern.ch/jalien/libjalieno2/-/merge_requests/3)
  
# What this PR does
I updated all affected files in O2 to explicitly use `std::` where needed for compilation to complete.
That is, this compilation command completed:
```bash
aliBuild build O2 --defaults o2
```

## Some smaller potential issues
- Some comments may be slightly misaligned due to code edits.
- I may have added `std::` in places where it wasn’t strictly necessary due to `using std::string;` declarations in the same `.cxx` files.
- Other, unknown to me, repositories might still rely on the polluted namespace and could break when the updated headers are published.

## A bit about me 🧑‍💻 
- Background in Software Engineering
- First experience working with C++ in a large codebase

Please let me know if anything is not how it should be and I'll fix it. (I do not know if I got the o2codechecker to run properly)

davidrohr reviewed (dismissed):
This looks correct to me, and indeed one should not have using statements in headers in the global namespace.

Please ignore the CI failure about code formatting, this is just because you touched files that are not formatted correctly. We can just force-merge it ignorning the CI errors.

Why did you mark the PR draft? In principle we should start the CI on it and check if it suceeds.

mrtineide commented:
> Why did you mark the PR draft? In principle we should start the CI on it and check if it suceeds.

Oh, that was just to play it safe. And also to maybe avoid the mass ping if there was something wrong with formatting and other smaller things. 

I'll mark it as ready.

davidrohr commented:
@mrtineide : CI looks good, could you please fix the merge conflict and update the PR. Then we can merge it.

mrtineide committed:
Add fully qualified names for std::string and std::map

The header TJAlienCredentials.h polluted the global namespace, included usually with CCDB
or something related.

Many source files in O2 took advantage of this.
This commit prepares for the removal of the polluted namespace.
The libraries/repo with TJAlienCredentials is JAliEn-ROOT and libjalieno2.
The problem was first noticed by @vkucera.

ktf commented:
Thank you!

vkucera commented:
Thanks a lot, @mrtineide !
